frontends/verilog/verilog_lexer.l

```diff
@@ -306,7 +306,7 @@ static bool isUserType(std::string &s)
 	return TOK_BASE;
 }
 
-<BASED_CONST>[0-9a-fA-FzxZX?][0-9a-fA-FzxZX?_]* {
+<BASED_CONST>[_]*[0-9a-fA-FzxZX?][0-9a-fA-FzxZX?_]* {
```

Comment:
Why not just removing the first square brackets? This should allow having the `_` as the first character as well in the based_const

```suggestion
<BASED_CONST>[0-9a-fA-FzxZX?_]* {
```

Reply:
But this would also allow to have ``TOK_BASED_CONSTVAL`` that is only ``_`` e.g.:
```
2'b_;
```
I double checked verilog standard and underscores isn't allowed as first character of a number (but vivado is parsing underscores as first characters and even allows to have only underscore constval):
```
The underscore character (_) shall be legal anywhere in a number except as the first character. The underscore character is ignored. 
```

I will follow the standard and update the code to do not use underscore as a first character.
Closing this PR.